nixos/modules/services/mail/goeland.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.goeland;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  options.services.goeland = {
+    enable = mkEnableOption (mdDoc "goeland");
+
+    settings = mkOption {
+      description = mdDoc ''
+        Configuration of goeland.
+        See the [example config file](https://github.com/slurdge/goeland/blob/master/cmd/asset/config.default.toml) for the available options.
+      '';
+      default = { };
+      type = types.submodule {
+        freeformType = tomlFormat.type;
+      };
+    };
+    schedule = mkOption {
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.goeland.settings.database = mkDefault "/var/lib/goeland/goeland.db";
```

Comment:
We should create the directories with tmpfiles.d

Reply:
It is a sqlite db that is supposed to be kept across reboots, is it still a good candidate for tmpfiles.d?

Reply:
tmpfiles.d can be used to create directories that hold data across reboots.

Reply:
Okay. Added it to the rules 👍🏽